File: socialuni-server/socialuni-user-sdk/src/main/java/com/socialuni/social/user/sdk/feignAPI/SocialuniMineUserAPI.java
package com.socialuni.social.user.sdk.feignAPI;

import com.socialuni.social.common.api.model.ResultRO;
import com.socialuni.social.common.api.model.user.SocialuniUserRO;
import com.socialuni.social.common.sdk.model.SocialuniImgAddQO;
import com.socialuni.social.user.sdk.model.QO.SocialPhoneNumQO;
import com.socialuni.social.user.sdk.model.QO.SocialProviderLoginQO;
import com.socialuni.social.user.sdk.model.QO.SocialUserEditQO;
import com.socialuni.social.user.sdk.model.QO.phone.SocialSendAuthCodeQO;
import com.socialuni.social.user.sdk.model.RO.login.SocialLoginRO;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;

import javax.validation.Valid;

@Tag(name = "用户模块/用户自身信息模块")
@FeignClient(name = "mineUser", url = "${socialuni.central-server-url:https://api.socialuni.cn}", path = "socialuni/mineUser")
public interface SocialuniMineUserAPI {
    @GetMapping("getMineUser")
    @Operation(summary = "获取用户个人详情")
    ResultRO<SocialuniUserRO> getMineUser();

    @PostMapping("editUser")
    @Operation(summary = "编辑用户")
    ResultRO<SocialuniUserRO> editUser(@RequestBody @Valid SocialUserEditQO socialUserEditQO);

    @PostMapping("addUserAvatarImg")
    @Operation(summary = "添加用户图片")
    ResultRO<SocialuniUserRO> addUserAvatarImg(@RequestBody @Valid SocialuniImgAddQO socialUserImgAddQO);

    //三方渠道登录，qq、wx、社交联盟，兼容各平台，h5、app、mp
    @PostMapping("providerLogin")
    ResultRO<SocialLoginRO<SocialuniUserRO>> providerLogin(@RequestBody @Valid SocialProviderLoginQO loginData);

    @PostMapping("phoneLogin")
    ResultRO<SocialLoginRO<SocialuniUserRO>> phoneLogin(@RequestBody @Valid SocialPhoneNumQO socialPhoneNumQO);

    @PostMapping("sendAuthCode")
    ResultRO<Void> sendAuthCode(@RequestBody @Valid SocialSendAuthCodeQO authCodeQO);
}

File: socialuni-js/packages/socialuni-api/src/api/SocialuniMineUserAPI.ts
import SocialuniUserRO from "../model/user/SocialuniUserRO";
import UserEditVO from "socialuni-sdk/src/model/user/UserEditVO"
import ImgAddQO from "socialuni-sdk/src/model/user/ImgAddQO"
import request from "socialuni-api/src/request/request";
import SocialSendAuthCodeQO from "socialuni-sdk/src/model/phone/SocialSendAuthCodeQO";
import UniProviderLoginQO from "socialuni-sdk/src/model/login/UniProviderLoginQO";
import SocialLoginRO from "socialuni-sdk/src/model/social/SocialLoginRO";
import SocialPhoneNumLoginQO from "socialuni-sdk/src/model/phone/SocialPhoneNumLoginQO";
import SocialuniMineUserRO from "../model/user/SocialuniMineUserRO";

export default class SocialuniMineUserAPI {
    static getMineUserInfoAPI() {
        return request.get<SocialuniUserRO>('socialuni/mineUser/getMineUser')
    }

    static editUserAPI(user: UserEditVO) {
        return request.post('socialuni/mineUser/editUser', user)
    }

    static addUserAvatarImgAPI(userImg: ImgAddQO) {
        return request.post<SocialuniUserRO>('socialuni/mineUser/addUserAvatarImg', userImg)
    }

    static sendAuthCodeAPI (phoneNum: string) {
        return request.post('socialuni/mineUser/sendAuthCode', new SocialSendAuthCodeQO(phoneNum))
    }
    static providerLoginAPI (loginData: UniProviderLoginQO) {
        return request.post<SocialLoginRO<SocialuniMineUserRO>>('socialuni/mineUser/providerLogin', loginData)
    }

    static socialuniPhoneLoginAPI (loginData: UniProviderLoginQO) {
        //接口已作废
        return request.post<SocialLoginRO<SocialuniMineUserRO>>('socialuni/mineUser/socialuniPhoneLogin', loginData)
    }

    static phoneLoginAPI (loginData: SocialPhoneNumLoginQO) {
        return request.post<SocialLoginRO<SocialuniMineUserRO>>('socialuni/mineUser/phoneLogin', loginData)
    }
}
